lib/spack/spack/environment/environment.py

```diff
@@ -62,8 +62,13 @@
 _active_environment = None
 
 
-#: path where environments are stored in the spack tree
-env_path = os.path.join(spack.paths.var_path, "environments")
+#: default path where environments are stored in the spack tree
+default_env_path = os.path.join(spack.paths.var_path, "environments")
+
+# ensure that this base directory exists since we require the root directory to be created
+# in our checks
+if not os.path.isdir(default_env_path):
+    os.mkdir(default_env_path)
```

Comment:
Why do we need the directory to exist if Spack isn't configured to use it?

Reply:
Ugh it has been a while. IIRC this is what was causing the `shell` tests in CI to fail. Spack does a check in the root environment directory to see if any environments are installed and if the directory is missing it hangs/fails.  I opted to always create to see if it fixed the failing CI tests.  Not exactly certain how deep this rabbit hole goes though.

Reply:
Addressed this as requested. Now we create the default location if is the `env_root_path()`

Reply:
Okay really addressed now.  The directory only gets created if it is used.